### DISCUSSION TITLE
How would an adapter send output to Console instead of REPL?

### DISCUSSION THREAD

**Discussion by bas-ie** [mfussenegger/nvim-dap#1127] [Q&A]
I'm aware there are a number of issues and discussion threads regarding this, and I think at this point I've read all of 'em! I know that some adapters support e.g. `console = 'integratedTerminal'`, and some don't. I realise that `nvim-dap` by default places all output events from the adapter into the REPL via https://github.com/mfussenegger/nvim-dap/blob/master/lua/dap/session.lua#L742-L748.

What I'm trying to understand is, how does an adapter send output somewhere other than the REPL? What event would it need to emit to make this work? I'm looking in particular at `lldb-dap`, which produces this kind of result when combined with `nvim-dap-ui` and a Rust target:

![ansi](https://github.com/mfussenegger/nvim-dap/assets/171905/a2b29d6c-4356-457e-8cae-20d2ce70fa53)

We can see here that lldb is emitting plain text, while the Rust console outputs have ANSI codes. Ideally these would not be sent to the REPL, as they just clutter things up and are difficult to read. I'm entirely willing to put in some work to modify `lldb-dap`'s behaviour if required, but I'd like to understand it first. Is the "debuggee" output (presumably stdout) also being sent to the REPL? How might the adapter control this?

Links to examples of adapters that do this well, or any other pointers gratefully accepted.

**Comment by pimbeenes** [comment]
Hi @bas-ie 

I just stumbled onto the same issue, did you end up finding a fix ?

**Reply by MironPascalCaseFan** [reply]
For me, lldb-dap starts streaming output in the console when I put `runInTerminal = true` flag in the configuration.

```lua
local dap = require("dap")
dap.configurations.rust = {
  {
    name = 'Launch',
    type = 'lldb',
    request = 'launch',
    program = require("dap.utils").pick_file,
    cwd = '${workspaceFolder}',
    stopOnEntry = false,
    args = {},
    runInTerminal = true,
  },
}

```

ref: https://github.com/llvm/llvm-project/blob/main/lldb/tools/lldb-dap/README.md

**Reply by pimbeenes** [reply]
This did not work for me on linux. Switching to codelldb fixed my problems.
